iot/iot-device-registry-infinispan/src/main/java/io/enmasse/iot/registry/infinispan/credentials/CacheCredentialService.java

```diff
@@ -0,0 +1,210 @@
+        get(tenantId, type, authId, null, span, resultHandler);
+    }
+
+    @Override
+    public void get(String tenantId, String type, String authId, JsonObject clientContext, Span span, Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler) {
+
+        var key = new CredentialsKey(tenantId, authId, type);
+
+        var future = cache
+                .getAsync(key)
+                .thenApplyAsync(r -> {
+                    if (r != null) {
+                        return CredentialsResult.<JsonObject>from(HTTP_OK, new JsonObject(r));
+                    } else {
+                        return CredentialsResult.<JsonObject>from(HTTP_NOT_FOUND);
+                    }
+                });
+
+        completeHandler(future, resultHandler);
+
+    }
+
+    @Override
+    public void add(String tenantId, JsonObject otherKeys, Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler) {
+
+        final String authId = otherKeys.getString(CredentialsConstants.FIELD_AUTH_ID);
+        final String type = otherKeys.getString(CredentialsConstants.FIELD_TYPE);
+
+        var key = new CredentialsKey(tenantId, authId, type);
+
+        add(key, otherKeys, resultHandler);
+
+    }
+
+    private void add(CredentialsKey key, JsonObject payload, Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler) {
+
+        var future = cache
+                .withFlags(Flag.FORCE_RETURN_VALUE)
+                .putIfAbsentAsync(key, payload.encode())
+                .thenApplyAsync(r -> {
+                    if (r == null) {
+                        return CredentialsResult.<JsonObject>from(HTTP_CREATED);
+                    } else {
+                        return CredentialsResult.<JsonObject>from(HTTP_CONFLICT);
+                    }
+                });
+
+        completeHandler(future, resultHandler);
+
+    }
+
+    @Override
+    public void update(String tenantId, JsonObject otherKeys, Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler) {
+
+        final String authId = otherKeys.getString(CredentialsConstants.FIELD_AUTH_ID);
+        final String type = otherKeys.getString(CredentialsConstants.FIELD_TYPE);
+
+        var key = new CredentialsKey(tenantId, authId, type);
+
+        update(key, otherKeys, resultHandler);
+
+    }
+
+    private void update(CredentialsKey key, JsonObject payload, Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler) {
+
+        var future = cache
+                .withFlags(Flag.FORCE_RETURN_VALUE)
+                .replaceAsync(key, payload.encode())
+                .thenApplyAsync(r -> {
+                    if (r == null) {
+                        return CredentialsResult.<JsonObject>from(HTTP_NOT_FOUND);
+                    } else {
+                        return CredentialsResult.<JsonObject>from(HTTP_NO_CONTENT);
+                    }
+                });
+
+        completeHandler(future, resultHandler);
```

Comment:
same issue :  the `futrure` variable may be null when the handler is completed ? 
also, why use `var` ?